extras/AFC_prep.py

```diff
@@ -64,95 +62,88 @@ def PREP(self, gcmd):
             extruders=json.load(open(self.AFC.VarFile + '.tool'))
 
         self.AFC.tool_cmds={}
+        self.buffer = {}
+        for PO in self.printer.objects:
+            if 'AFC_buffer' in PO:
+                tmpBUFFER = self.printer.lookup_object(PO)
+                self.buffer[tmpBUFFER.name] = tmpBUFFER
+        hub = []
+        for PO in self.printer.objects:
+            if 'AFC_hub' in PO:
+                hub.append(PO.split()[-1])
 
         for PO in self.printer.objects:
             if 'AFC_stepper' in PO and 'tmc' not in PO:
                 LANE=self.printer.lookup_object(PO)
-                self.AFC.stepper[LANE.name]=LANE
+                UNIT=self.printer.lookup_object('AFC_unit '+ LANE.unit)
+                if LANE.name not in UNIT.lanes: UNIT.lanes.append(LANE.name)    #add lanes to units list
+                self.AFC.stepper[LANE.name]=LANE                                #add list of all lanes
 
-                # If extruder section exists in vars file add currently stored data to AFC.extruders array
```

Comment:
With the removal of setting `self.AFC.extruders` when `system_test` is called it errors out. Not sure why some of the stuff was redone in this file.

![image](https://github.com/user-attachments/assets/13976c24-f1e6-4e0e-a723-519ffdba042b)

Reply:
Weird indoor ger that error 

Look when I get home

Reply:
Ir was done for all the reasons of the last week